### PR TITLE
setup javascript mono-repo tooling

### PR DESCRIPTION
**Pull request by fuxingloh**

### CODE REVIEW

**Review by joeldavidw — APPROVED** [review]

File: packages/salmon-core/package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
```
Comment (monstrobishi) [inline packages/salmon-core/package.json:1]: so the existing salmon will live in here?

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+  preset: 'ts-jest',
+  testRegex: '((\\.|/)(e2e|test|spec))\\.[jt]sx?$',
+  moduleNameMapper: {
+    '@defichain/salmon-(.*)': '<rootDir>/packages/salmon-$1/src'
```
Comment (monstrobishi) [inline jest.config.js:5]: interesting, this wildcard takes care of the copy pasting

File: tsconfig.build.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  "compilerOptions": {
+    "composite": true,
+    "declaration": true,
+    "declarationMap": true,
+    "sourceMap": true
+  },
+  "files": [],
+  "references": [
+    {
+      "path": "./packages/salmon-core/tsconfig.build.json"
```
Comment (monstrobishi) [inline tsconfig.build.json:12]: does this support the wildcard too?

**Review by monstrobishi — APPROVED** [review]

File: packages/salmon-core/package.json (hunk 1 above)
Comment (fuxingloh) [reply packages/salmon-core/package.json:1]: Yup, I am going to refactor it into separated modules. 3 of us will work on it so we have good coverage of responsibilities of `salmon-*` and further break down the downstream implementation of the adapter with engineers from jellyfish.

File: tsconfig.build.json (hunk 3 above)
Comment (fuxingloh) [reply tsconfig.build.json:12]: I think it doesn't support them, but I am experimenting with not including them in the project. Might not need them.

**Pull request merged**